### PR TITLE
Add admin login page

### PR DESCRIPTION
**Pull request by enricodelarosa**
## Change Type
*Put an `x` in all the boxes that apply
- [] Bugfix (non-breaking change which fixes an issue)
- [x] Feature (non-breaking change which adds functionality)
- [] Breaking change (fix or feature that would cause existing functionality to change)

## Linear issue link
https://linear.app/oku-no-kumo/issue/OKU-25/admin-login-page

## Before & after screenshots or video recordings (Before & After)

image

image

### CODE REVIEW

File: project_app/components/AdminLayout.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+export default function AdminLayout(props) {
+  return (
+    <div className={styles.adminLayout}>
+      <Image
+        src="/logo.svg"
+        width={227.77}
+        height={150}
+        left={50}
+        right={50}
```
Comment (idiglove) [inline project_app/components/AdminLayout.js:13]: lets have this in as a styled components

File: project_app/components/AdminLogin.js [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+        body: JSON.stringify({
+          emailOrUsername: emailOrUsername,
+          password: password,
+        }),
+      }
+    );
+
+    const resBody = await response.json();
+
+    console.log(resBody);
```
Comment (idiglove) [inline project_app/components/AdminLogin.js:30]: remove console log

File: project_app/components/AdminLogin.js [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({
+          emailOrUsername: emailOrUsername,
+          password: password,
+        }),
+      }
+    );
+
+    const resBody = await response.json();
```
Comment (idiglove) [inline project_app/components/AdminLogin.js:28]: much better if we make resBody as a state

File: project_app/styles/AdminLayout.module.css [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+}
+
+.adminContent {
+  flex-grow: 1;
+  width: 100%;
+  display: flex;
+  justify-content: center; /* centers content horizontally */
+  align-items: center; /* centers content vertically */
+}
+
```
Comment (idiglove) [inline project_app/styles/AdminLayout.module.css:16]: we use styled components in this project.
pls contact @twentymurial33 if you need help

**Review by idiglove — CHANGES_REQUESTED** [review]
I have some comments

File: project_app/styles/AdminLayout.module.css (hunk 4 above)
Comment (twentymurial33) [reply project_app/styles/AdminLayout.module.css:16]: @enricodelarosa we have the `AdminLayout` component now using Styled Components 👍

File: project_app/components/AdminLogin.js [hunk 5]
```diff
@@ -0,0 +1,185 @@
+import { useState, useContext, useEffect } from "react";
+import { GlobalStateContext } from '../pages/_app';
+import styled from "styled-components";
+//import styles from "../styles/AdminLogin.module.css";
```
Comment (idiglove) [inline project_app/components/AdminLogin.js:4]: remove comments

File: project_app/components/AdminLogin.js [hunk 6]
```diff
@@ -0,0 +1,185 @@
...
+import { useRouter } from 'next/router';
+
+export default function AdminLogin() {
+    const router = useRouter();
+    const { token, setToken } = useContext(GlobalStateContext);
+    const [emailOrUsername, setEmailOrUsername] = useState("");
+    const [password, setPassword] = useState("");
+
+    useEffect(() => {
+        console.log(token);
```
Comment (idiglove) [inline project_app/components/AdminLogin.js:20]: remove console log

File: project_app/components/AdminLogin.js [hunk 7]
```diff
@@ -0,0 +1,185 @@
...
+        if (token != null) {
+            // Verify that token is not tampered with using api
+            router.push('adminpage');
```
Comment (idiglove) [inline project_app/components/AdminLogin.js:23]: what is the diff of admin and adminpage?

File: project_app/components/AdminLogin.js [hunk 8]
```diff
@@ -0,0 +1,185 @@
...
+                    <CFormFloating style={{ marginBottom: '1rem' }}>
+                        <CFormInput
+                            type="email"
+                            id="emailOrUsername"
+                            placeholder="name@example.com"
+                            value={emailOrUsername}
+                            onChange={(e) => {
+                                setEmailOrUsername(e.target.value);
+                            }}
+                            style={{ backgroundColor: "#2B2B2B", color: "#FFFFFF" }}
```
Comment (idiglove) [inline project_app/components/AdminLogin.js:83]: can we have these styles in styledcomponent?

File: project_app/pages/adminpage.tsx [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+export default function admin() {
+    const router = useRouter();
+    const { token } = useContext(GlobalStateContext);
+
+    useEffect(() => {
+        if (token == null) {
+            router.push('/admin');
+        } else {
+
+        }
```
Comment (idiglove) [inline project_app/pages/adminpage.tsx:16]: whats in this else?

**Review by idiglove — CHANGES_REQUESTED** [review]

File: project_app/pages/adminpage.tsx (hunk 9 above)
Comment (enricodelarosa) [reply project_app/pages/adminpage.tsx:16]: Actually nothing hehe, will be removing it.

File: project_app/components/AdminLogin.js (hunk 7 above)
Comment (enricodelarosa) [reply project_app/components/AdminLogin.js:23]: Admin is the admin login page and admin page will be what an admin sees after loggin in

File: project_app/components/AdminLogin.js (hunk 8 above)
Comment (enricodelarosa) [reply project_app/components/AdminLogin.js:83]: This is what I actually wanted to do but I had problems with styling existing components, but a quick google search yielded results <3

**Review by idiglove — APPROVED** [review]
the only thing is the fonts are not properly styled. but lets do it on next sprint

**Pull request merged**